pkg/networkservice/common/timeout/server.go

```diff
@@ -60,17 +59,20 @@ func NewServer(onTimout *networkservice.NetworkServiceServer) networkservice.Net
 }
 
 func (t *timeoutServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
-	ct, err := t.createTimer(ctx, request)
+	conn, err := next.Server(ctx).Request(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	ct, err := t.createTimer(request.GetConnection())
```

Comment:
As pointed out by @Bolodya1997 [here](https://github.com/networkservicemesh/sdk/pull/520#issue-502056572) we should be computing our timer based on the returned connection.